.github/workflows/check-semver.yml

```diff
@@ -0,0 +1,23 @@
+name: Check Semver
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  check-semver:
+    name: Check semver
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          # Check out the merge commit as default.
+          # Reference: actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit
+          ref: ${{ github.event.pull_request.head.ref }}
+          # Check out from the source (forked repository) rather than the destination.
+          # This helps ensure we are working with the correct source code.
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment:
With the `pull_request` target, I think you probably don't need these anymore. The arguments here set the values that are already set by default, I believe.
```suggestion
```

Reply:
c0a911b3e92570731e3e7ff632b9dc470ed684f0